### PR TITLE
General github tests adoption

### PR DESCRIPTION
**Pull request by zilm13**
Restored

Moving to new tests from https://github.com/ethereum/eth2.0-specs/tree/dev/test_generators
Everything done and passed except 2 operations/attestations tests.

Keep `WIP` until we have repository with automatically built `yaml` tests, plus `gradle` test task should be updated to match this changes. Currently it's done manually.

### CODE REVIEW

**Comment by zilm13** [comment]
Everythng is passed.  On hold until generated tests repository is online.

**Comment by zilm13** [comment]
There are a lot of things broken after merging v0.6.1:
1) 2 tests in BLS, test generator is based on 1.6.0 version of py-ecc  which was released in March and based on keccak (now changed to sha256)
2) 2 valid operation/attestations tests. In research
3) some SSZ static tests. In research.

Everything else is passed.

File: start/common/src/main/resources/config/spec-constants.yml [hunk 1]
```diff
@@ -20,11 +20,13 @@ specConstants:
   miscParameters:
     SHARD_COUNT: 1024
     TARGET_COMMITTEE_SIZE: 128
+    MAX_BALANCE_CHURN_QUOTIENT: 32
```
[the remark this answers is not in the document]
Comment (zilm13) [reply start/common/src/main/resources/config/spec-constants.yml]: just haven't included these files in merge

File: test/src/test/java/org/ethereum/beacon/test/StateTestUtils.java [hunk 2]
```diff
@@ -204,19 +204,28 @@ public static MutableBeaconState parseBeaconState(SpecConstants specConstants, B
     state.setLatestEth1Data(parseEth1Data(data.getLatestEth1Data()));
     state.setDepositIndex(UInt64.valueOf(data.getDepositIndex()));
 
+    state.getValidatorRegistry().clear();
     state.getValidatorRegistry().addAll(parseValidatorRegistry(data.getValidatorRegistry()));
-    state.getBalances().addAll(parseBalances(data.getValidatorBalances()));
+    state.getBalances().clear();
```
[the remark this answers is not in the document]
Comment (zilm13) [reply test/src/test/java/org/ethereum/beacon/test/StateTestUtils.java]: done

File: test/src/test/java/org/ethereum/beacon/test/runner/ssz/SszStaticRunner.java [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    return fromInfo(valueTypeInfo);
+  }
+
+  public Optional<String> run() {
+    Class valueType = findSpecClassByName(testCase.getTypeName());
+    Object fromSerialized =
+        sszSerializer.decode(BytesValue.fromHexString(testCase.getSerialized()), valueType);
+    Object simplified = objectSerializers.get(valueType).map(fromSerialized);
+    ObjectNode expectedValue = yamlMapper.convertValue(testCase.getValue(), ObjectNode.class);
+    // XXX: expected goes second as our constructed node contains with overridden `equals` operators
```
[the remark this answers is not in the document]
Comment (zilm13) [reply test/src/test/java/org/ethereum/beacon/test/runner/ssz/SszStaticRunner.java]: You couldn't guarantee type of numeric JsonNode for example, 0 could go as `LongNode` or as `IntNode`, and `IntNode` with the same number underneath doesn't equal `LongNode`. I have not found any better solution to make this things correctly compared.

File: test/src/test/java/org/ethereum/beacon/test/runner/state/StateRunner.java [hunk 4]
```diff
@@ -1,103 +1,82 @@
...
 
-  private BeaconStateEx buildInitialState(BeaconChainSpec spec, BeaconStateData stateData) {
-    BeaconState state = StateTestUtils.parseBeaconState(spec.getConstants(), stateData);
-    return new BeaconStateExImpl(state, TransitionType.BLOCK);
+  private void processDeposit(Deposit deposit, BeaconState state) {
+    try {
+      DepositVerifier depositVerifier = new DepositVerifier(spec);
+      assert depositVerifier.verify(deposit, state).isPassed();
+      spec.process_deposit((MutableBeaconState) state, deposit);
+    } catch (Exception | AssertionError ex) {
```
[the remark this answers is not in the document]
Comment (zilm13) [reply test/src/test/java/org/ethereum/beacon/test/runner/state/StateRunner.java]: Please. check the way I've done it.

**Comment by zilm13** [comment]
@mkalinin fixed everything according to your review

File: test/src/test/java/org/ethereum/beacon/test/runner/ssz/SszStaticRunner.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (mkalinin) [reply test/src/test/java/org/ethereum/beacon/test/runner/ssz/SszStaticRunner.java]: gotcha!

File: consensus/src/main/java/org/ethereum/beacon/consensus/verifier/operation/AttestationVerifier.java [hunk 5]
```diff
@@ -27,8 +28,9 @@ public VerificationResult verify(Attestation attestation, BeaconState state) {
     try {
       spec.verify_attestation(state, attestation);
       return VerificationResult.PASSED;
-    } catch (Exception e) {
-      return VerificationResult.failedResult(e.getMessage());
+    } catch (SpecCommons.SpecAssertionFailed e) {
+      String error = e.getStackTrace().length > 0 ? e.getStackTrace()[1].toString() : "SpecAssertion";
```
Comment (mkalinin) [inline consensus/src/main/java/org/ethereum/beacon/consensus/verifier/operation/AttestationVerifier.java]: Literally, `e.getStackTrace().length > 0` check doesn't guarantee that `e.getStackTrace()[1]` won't turn with index out of bounds; but I believe it's OK for this particular case.

Do you mind overriding `getMessage` for `SpecAssertionFailed` which would execute this line of code and as a result return `error` string?

File: test/src/test/java/org/ethereum/beacon/test/runner/state/StateRunner.java [hunk 6]
```diff
@@ -1,103 +1,83 @@
...
 
-  private BeaconStateEx buildInitialState(BeaconChainSpec spec, BeaconStateData stateData) {
-    BeaconState state = StateTestUtils.parseBeaconState(spec.getConstants(), stateData);
-    return new BeaconStateExImpl(state, TransitionType.BLOCK);
+  private void processDeposit(Deposit deposit, BeaconState state) {
+    try {
+      DepositVerifier depositVerifier = new DepositVerifier(spec);
+      assert depositVerifier.verify(deposit, state).isPassed();
+      spec.process_deposit((MutableBeaconState) state, deposit);
+    } catch (SpecCommons.SpecAssertionFailed | AssertionError ex) {
```
Comment (mkalinin) [inline test/src/test/java/org/ethereum/beacon/test/runner/state/StateRunner.java]: We don't use `assert` in the spec, do we? I am thinking of some rare cases when `AssertionError` has been due to odd state of some object (usual way `assert` is made for); in that case I'd rather prefer to see this exception than get it swallowed here.

**Comment by zilm13** [comment]
Update. Currently failed:
- `BLS/testBlsSignMessage`, `BLS/testBlsMessageHashCompressed`, requires update of hash method in test generator according to spec, issue created: https://github.com/ethereum/py_ecc/issues/68 . After resolving generator should be updated too.
- `ssz_static` - requires SSZ SOS implementation, blocked by PR resolving #124 
- `operations/attestations` - there were a lot changes in attestations, generator should be updated first. Job started but not yet finished. Plus it will be updated not for v0.6.1, but for current spec, so should be ignored at this moment.

All tests are blocked by resolving of this issue: https://github.com/ethereum/eth2.0-spec-tests/issues/1
current repository does not contain spec configs.

**Comment by zilm13** [comment]
> ssz_static - requires SSZ SOS implementation, blocked by PR resolving #124
resolved by #132 , tests passed

**Pull request closed without merging**